src/kernels/variables/JupyterVariablesProvider.ts

```diff
@@ -45,6 +46,15 @@ export class JupyterVariablesProvider implements NotebookVariableProvider {
         }
     }
 
+    private _getVariableResultCacheKey(notebookUri: string, parent: Variable | undefined, start: number) {
```

Comment:
these could stay in the cache class and be overridden by the implementations

Reply:
@amunger I was thinking about that originally but then later on found that we need both `parent` and `start` for generating cache key, but for generating summary, only `parent/variable` is needed, thus I went with moving the cache key generation out of the cache class. Hope this makes sense.